docs/rules/no-reaching-inside.md

```diff
@@ -0,0 +1,64 @@
+# no-reaching-inside
+
+Use this rule to prevent importing the submodules of other modules.
+
+## Rule Details
+
+This rule has one option, `allow` which is an array of minimatch patterns to identify directories whose children can be imported explicitly.
```

Comment:
Can you make `minimatch patterns` a link that links to documentation about minimatch patterns?